spec/mailers/user_mailer_spec.rb

```diff
@@ -67,36 +106,56 @@
     end
   end
 
-  describe :issue_update do
-    context :delayed_job do
-      before do
-        # Delayed Job does not preserve the closure, so the context of the
-        # delayed method call does not contain the user anymore, who triggered
-        # the job. Instead, the anonymous user is returned.
-        User.current = User.anonymous
+  describe :work_package_added do
+    before do
+      UserMailer.work_package_added(recipient, work_package, user).deliver
+    end
 
-        UserMailer.work_package_updated(user, journal, user)
-      end
+    it_behaves_like 'mail is sent'
 
-      it { expect(User.current).to eq(user) }
+    it 'contains the WP subject in the mail subject' do
+      expect(ActionMailer::Base.deliveries.first.subject).to include(work_package.subject)
+    end
 
-      after do
-        User.current = User.anonymous
-      end
+    it_behaves_like 'does only send mails to author if permitted'
+  end
+
+  describe :work_package_updated do
+    before do
+      UserMailer.work_package_updated(recipient, journal, user).deliver
     end
+
+    it_behaves_like 'mail is sent'
+
+    it_behaves_like 'does only send mails to author if permitted'
+  end
+
+  describe :wiki_content_added do
+    let(:wiki_content) { FactoryGirl.create(:wiki_content) }
+
+    before do
+      UserMailer.wiki_content_added(recipient, wiki_content, user).deliver
+    end
+
+    it_behaves_like 'mail is sent'
+
+    it_behaves_like 'does only send mails to author if permitted'
   end
 
   describe :wiki_content_updated do
     let(:wiki_content) { FactoryGirl.create(:wiki_content) }
-    let!(:mail) { UserMailer.wiki_content_updated(user, wiki_content) }
 
-    it 'should link to the latest version diff page' do
-      expect(mail.body.encoded).to include 'diff/2'
+    before do
+      UserMailer.wiki_content_updated(recipient, wiki_content, user).deliver
     end
 
-    after do
-      User.current = nil
+    it_behaves_like 'mail is sent'
+
+    it 'should link to the latest version diff page' do
+      expect(ActionMailer::Base.deliveries.first.body.encoded).to include 'diff/2'
     end
+
+    it_behaves_like 'does only send mails to author if permitted'
```

Comment:
This should be covered by `work_package_updated`...

But you told me in a discussion that you received unwanted mails for comments, but nto for updates... so I will have another look...